seed/php-sdk/imdb/clientName/src/Core/Client/RawClient.php

```diff
@@ -31,23 +32,41 @@ class RawClient
      *   baseUrl?: string,
      *   client?: ClientInterface,
      *   headers?: array<string, string>,
+     *   maxRetries?: int,
      * } $options
      */
     public function __construct(
         public readonly ?array $options = null,
     ) {
-        $this->client = $this->options['client'] ?? new Client();
+        $this->client = $this->options['client']
+            ?? $this->createDefaultClient();
         $this->headers = $this->options['headers'] ?? [];
     }
 
     /**
+     * @return Client
+     */
+    private function createDefaultClient(): Client
+    {
+        $handler = HandlerStack::create();
+        $handler->push(RetryMiddleware::create());
+        return new Client(['handler' => $handler]);
+    }
```

Comment:
We'll want to wrap the user-defined `ClientInterface` with retry logic, too. Can we refactor this so that we resolve the `ClientInterface` then wrap it after the fact?

Reply:
I would, but it's not possible. If the user provides their own client, they need to create their own stack and add the middleware themselves.